### ISSUE TITLE
Accessibility warning

### ISSUE THREAD

**Issue by bsylwek** [jmcameron/attachments#121]
Hello
Users report a problem with a warning in the accessibility of the page in connection with downloading attachments. This is very important for people with disabilities. I have attached a screenshot showing it. I managed to locate that this is a problem in the Attachements template in the localization

components/com_attachments/tmpl/attachments/default.php

this code?

`$a_class = 'at_icon';
			$show_link = "<a class=\"". $a_class . "\" href=\"$url\"$target title=\"$tooltip\" alt=\"$tooltip\">";
			$show_link .= HTMLHelper::image('com_attachments/file_icons/'.$icon, $tooltip, null, true);
		}
`

![Image](https://github.com/user-attachments/assets/d77016af-5153-4792-8e29-efb5bfb8d3a5)

How can I solve this problem?

**Comment by JLTRY** [comment]
@bsylwek what do you want as alt text depending on the attachment file ? can you give an example ?

**Comment by bsylwek** [comment]
If I understand correctly, this accessibility warning should be described in the alt and or title, to which page it directs? I think it is the attachment page defined in the alt and or ttile field? Problem only concerns the icon of a file, e.g. pdf.gif. People with disabilities pay attention to this on their content readers.

**Comment by JLTRY** [comment]
do you have the url of the screenshot, so that I can check the accessibility
do you use a specific tool to check the accessibility ?

**Comment by bsylwek** [comment]
Yes, please scroll bottom and click on the right, a round icon of human availability and show settings all to ON and this warning will be displayed
https://www.bip.dps2.vernet.pl/dostep-do-informacji-niezamieszczonej-w-bip

**Comment by bsylwek** [comment]
See I removed this line of code and everything is fine.

![Image](https://github.com/user-attachments/assets/e9b75d52-c57f-43c0-b128-3872e1c22bc0)

`$show_link = "<a class=\"". $a_class . "\" href=\"$url\"$target title=\"$tooltip\" alt=\"$tooltip\">";
`

**Comment by parapente** [comment]
Yes but now if you happen to click on the icon the file will not download. You basically removed the link that was the reason for the warning. That does not make it the right thing to do.

**Comment by bsylwek** [comment]
I'm wondering how to solve this differently. I can't find instructions anywhere on how to make the link next to the icon and not get an error.

**Comment by JLTRY** [comment]
Could you share the name of the tool used to check the accessibility ?

**Comment by bsylwek** [comment]
This accessibility validator also shows an error here.
[](https://www.accessibilitychecker.org/audit/?website=https%3A%2F%2Fwww.bip.dps2.vernet.pl%2Fdostep-do-informacji-niezamieszczonej-w-bip&flag=eu  )

**Comment by parapente** [comment]
That is a different error: "Ensure touch target have sufficient size and space" and it is actually wrong as the icon is not the only target for downloading the file. Also if you want you can improve the icon or link size by using some css in your template adding more padding to the anchor.

**Comment by JLTRY** [comment]
@bsylwek can you please check at my side it is ok

**Comment by bsylwek** [comment]
I noticed you did a component update? I guess that didn't change anything. I updated.

[https://github.com/jmcameron/attachments/tree/b0281dbf9280840305120324a7aff2dada4d672c](url)

**Comment by JLTRY** [comment]
@bsylwek no component update a source update that should be checked
Here is the raw format of the updated file 
https://raw.githubusercontent.com/JLTRY/attachments/refs/heads/master/attachments_component/site/tmpl/attachments/default.php

**Comment by bsylwek** [comment]
I uploaded it. It works for me. I'll check it with different accessibility validators.

**Comment by JLTRY** [comment]
please open an other issue if more changes are needed
